### ISSUE TITLE
Loans: Unify errors in the pallet level

### ISSUE THREAD

**Issue by lemunozm** [centrifuge/centrifuge-chain#1436]
### Description

Currently, we have split errors in `pallet-loans` depending on its type (see [here](https://github.com/centrifuge/centrifuge-chain/blob/b006fb86cd0a3121ffc2ec603436c7b678c81a4a/pallets/loans/src/types/mod.rs#L31)). Nevertheless, it's only because of code organization and the message is not containing the last variant level of the error, making the debugging phase harder. So I propose to unify all errors as usual in the pallet `Error` level.

Example of error message:
```sh
Err(Module(ModuleError { index: 9, error: [12, 1, 0, 0], message: Some("BorrowLoanError") }))
```
We lost what type of `BorrowLoanError` error it is. Is it `MaxAmountExceeded`? or `Restriction`? or `MaturityDatePassed`?

**Comment by lemunozm** [comment]
WDYT @mustermeiszer @offerijns ?

**Comment by lemunozm** [comment]
Also, sometimes grouping them by "action" means that they can not be reused out from that action, and it is difficult to know which error should be returned. For example, a method of external pricing can be used both for borrowing and repaying, and it can generate errors, should it be a `BorrowLoanError` or `RepayLoanError`? We do not know inside of the pricing method.

**Comment by mustermeiszer** [comment]
I actually like that grouping. I think it makes the error reading in code at least easier. Couldn't we write a `fmt` for the enum and make it readable again? 

> Re: Actions

Woudln't it be a `PricingError` and the user knows that it is happeing in an `Borrow`-action?

**Comment by offerijns** [comment]
I think for this:
> EDIT: it can be known looking the second index (1) which represents the variant position, although it's still non-intuitive 🤔
We should keep it as is. The UI at some point can figure it out based on the second index, and show the actual error based.

**Comment by lemunozm** [comment]
> Couldn't we write an fmt for the enum and make it readable again?

I'll test it 👍🏻. Not sure how Substrate gets the name of the error TBH.

**Comment by lemunozm** [comment]
> Woudln't it be a PricingError and the user knows that it is happeing in an Borrow-action?

Yes, probably now it makes more sense to change it to another error type of organization. I mean, the user always knows it's a borrow or repay error because they are calling the extrinsic with that name, so no extra information here. It makes much more sense to move the error types per module. Also, each module could contain its own error there. I like this, if you agree I can change the organization to this way.

**Comment by mustermeiszer** [comment]
> I'll test it 👍🏻. Not sure how Substrate gets the name of the error TBH.

You should expand a pallet once and check there.

**Comment by lemunozm** [comment]
To summarize this issue: errors should still be grouped with sub-error types, but instead of grouping them by action, they should be grouped per module. i.e. `PricingError`, `InterestError`, `ValuationError`...
